src/audio/pipeline/pipeline-schedule.c

```diff
@@ -79,7 +79,7 @@ static enum task_state pipeline_task_cmd(struct pipeline *p,
 			p->status = COMP_STATE_PAUSED;
 		}
 
-		if (err == PPL_STATUS_PATH_STOP) {
+		if (err == PPL_STATUS_PATH_STOP && !p->trigger.aborted) {
```

Comment:
Can we have a comment here on why we check both err and aborted.

Reply:
@lgirdwood thanks for checking this. The first 6 commits in this draft are actually from #5174, they'll be dropped here once that one is merged. As for the rest - this is still very much WiP. E.g. this specific code path will probably go when PAUSE and STOP are merged, or at least it will be heavily modified.